policy/auto-join.json

```diff
@@ -43,43 +43,75 @@
                 },
-                "resourceGroupName": "[parameters('fleet-resource-group')]",
                 "deployment": {
                     "properties": {
                         "mode": "incremental",
                         "template": {
                             "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
                             "contentVersion": "1.0.0.0",
                             "parameters": {
-                                "fullDbName": {
-                                    "type": "string"
+                                "fleet-name": {
+                                  "type": "string"
+                                },
+                                "fleet-resource-group": {
+                                  "type": "string"
+                                },
+                                "fleet-subscription-id": {
+                                  "type": "string"
+                                },
+                                "cluster-resource-id": {
+                                  "type": "string"
                                 }
                             },
                             "resources": [
                                 {
-                                    "apiVersion": "2024-04-01",
-                                    "type": "Microsoft.ContainerService/fleets/members",
-                                    "name": "[concat(parameters('fleet-name'), '/', parameters('member-name'))]",
-                                    "properties": {
-                                        "clusterResourceId": "[parameters('cluster-resource-id')]",
-                                        "group": "auto"
+                                  "name": "[concat('join-with-fleet-', parameters('fleet-name'))]",
+                                  "type": "Microsoft.Resources/deployments",
+                                  "apiVersion": "2022-09-01",
+                                  "subscriptionId": "[parameters('fleet-subscription-id')]",
+                                  "resourceGroup": "[parameters('fleet-resource-group')]",
+                                  "properties": {
+                                    "mode": "Incremental",
+                                    "template": {
+                                      "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                                      "contentVersion": "1.0.0.0",
+                                      "resources": [
+                                        {
+                                          "type": "Microsoft.ContainerService/fleets/members",
+                                          "apiVersion": "2024-05-02-preview",
+                                          "name": "[concat(parameters('fleet-name'), '/', parameters('member-name'))]",
+                                          "properties": {
```

Comment:
it would be great if we could use a property of the found cluster, to set the member.group property

Reply:
for example, can we automatically set the member.group property to the cluster's location?

Reply:
Done.  I included an optional parameter for looking up the tags on the aks cluster and grabbing the value to be used as the group.